crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -793,15 +796,30 @@ impl LogsControllerInner {
                         },
                     );
                 }
-                // todo how to figure out whether a previous logs controller instance has started
-                //  sealing a loglet? Can we ask the log servers about it? Should we periodically
-                //  check via bifrost?
             }
         }
 
         Ok(())
     }
 
+    fn on_logs_tail_updates(&mut self, updates: &LogsTailUpdates) {
+        for (log_id, state) in self.logs_state.iter_mut() {
+            let Some(update) = updates.get(log_id) else {
+                // no updates for this log
+                continue;
+            };
+
+            let TailState::Sealed(seal_lsn) = update.tail else {
+                // if tail is open, we don't have to do anything
+                // because in all cases, the `LogState` here is
+                // the one that should remain in effect
+                continue;
+            };
```

Comment:
Nit: Given that in most cases many tail states will be `TailState::Open`, we could safe a few CPU cycles by iterating over `updates` and looking up the state in `self.logs_state` if the state is `TailState::Sealed`.